Use dark theme in Juno

MikeInnes opened the pull request:
If the user has not overridden the empty default style, we use gadfly_dark in Juno.

Modifying the plot feels unclean to me, so there may be a nicer way to do that – but I don't think it should cause issues if not.

spencerlyon2 commented:
Hmm. I see what you're saying about mutating the plot. The issue here is that after the plot is displayed in the juno pane, the style is changed to this and users won't be able to save the figure with the default style. I'd like to figure out a way around that...

MikeInnes commented:
Actually in this case I'm preserving the original `style` property and putting it back after we're done displaying. Still not totally clean but I don't think it should have that specific issue.

spencerlyon2 commented:
ha, I didn't even catch that! Good work.

I agree that it isn't totally clean, but I can't think of a better way to accomplish this.

I'll leave this open for a couple hours in case someone has an idea, otherwise I'll merge.

spencerlyon2 commented:
Oh wait, I'm wrong about that. 

The way saving a figure works is the plot is displayed and then Blink asks plotly.js for the associated svg data. In this case I'm pretty sure changing the display is the same as changing the saved figure.

MikeInnes commented:
Ok, I changed this so that it just sets a default at startup.

I also moved the initialisation of DEFAULT_STYLE to __init__, since I'm guessing that's not supposed to be fixed at precompile time.
